kotlin/create-and-edit-mobile-geodatabases/src/main/java/com/esri/arcgisruntime/sample/createandeditmobilegeodatabases/MainActivity.kt

```diff
@@ -0,0 +1,191 @@
+            sharingIntent.type = "*/*"
+            sharingIntent.putExtra(Intent.EXTRA_STREAM, Uri.parse(geodatabase?.path))
+            startActivity(Intent.createChooser(sharingIntent, "Share using"))
+        }
+    }
+
+    private fun createGeodatabase() {
+        // define the name of the geodatabase file
+        val file = File(getExternalFilesDir(null)?.path + "/locationHistory.geodatabase")
+        if (file.exists()) file.delete()
+        // create the geodatabase file
+        val geodatabaseFuture = Geodatabase.createAsync(file.path)
+        geodatabaseFuture.addDoneListener {
+            Log.e(TAG, "Geodatabase created")
+            geodatabase = geodatabaseFuture.get()
+
+            // construct a table description
+            val tableDescription =
+                TableDescription(
+                    "LocationHistory",
+                    SpatialReferences.getWgs84(),
+                    GeometryType.POINT
+                )
+            // add fields to the table
+            tableDescription.fieldDescriptions.addAll(
+                listOf(
+                    FieldDescription("oid", Field.Type.OID),
+                    FieldDescription("name", Field.Type.TEXT),
+                    FieldDescription("sequence", Field.Type.INTEGER),
+                    FieldDescription("collection_timestamp", Field.Type.DATE)
+                )
+            )
+
+            // set some table properties
+            tableDescription.setHasAttachments(false)
+            tableDescription.setHasM(false)
+            tableDescription.setHasZ(false)
+
+            val tableFuture = geodatabaseFuture.get().createTableAsync(tableDescription)
+            setupMapFromGeodatabase(tableFuture)
+        }
+    }
+
+    private fun setupMapFromGeodatabase(tableFuture: ListenableFuture<GeodatabaseFeatureTable>) {
+        tableFuture.addDoneListener {
+            val table = geodatabase?.geodatabaseFeatureTables?.first {
+                it.tableName == "LocationHistory"
+            }
+            val featureLayer = FeatureLayer(table)
+            mapView.map.operationalLayers.add(featureLayer)
+            this.featureTable = table
+        }
+    }
+
+    /**
+     * Adds a feature to the feature table on map click
+     */
+    private fun addFeature(mapPoint: Point) {
+        // set up the feature attributes
+        val featureAttributes = mutableMapOf<String, Any>()
+        featureAttributes["name"] = "User's location"
+        featureAttributes["sequence"] = sequence
+        featureAttributes["collection_timestamp"] = Date().time
+
+        // create a new feature
+        val feature = featureTable?.createFeature(featureAttributes, mapPoint)
+        // add the feature to the feature table
+        featureTable?.addFeatureAsync(feature)?.addDoneListener {
+            // feature added to the table
+            Log.e(TAG, "Added feature to table, size = ${featureTable!!.totalFeatureCount}")
```

Comment:
Here we have a sample app that creates a geodatabase and adds features to the table. 
Issue:  
I'm not able to see the feature being added to the table here. The total feature count returns 0 always.